flax/nn/__init__.py

```diff
@@ -29,3 +29,6 @@
 from .recurrent import LSTMCell, GRUCell, ConvLSTM, OptimizedLSTMCell
 from .stochastic import make_rng, stochastic, dropout, is_stochastic
 # pylint: enable=g-multiple-import
+import warnings
+warnings.filterwarnings("default", category=DeprecationWarning, module=__name__)
+warnings.warn("the nn module is deprecated, use linen instead", DeprecationWarning)
```

Comment:
How does this work in Python? If someone does `import flax.nn` it will execute this code. But what if someone does `import flax.nn.base`?

Reply:
I lost the reference, but yeah if import anything from it it shows a deprecation warning